### ISSUE TITLE
Add kernel default parameters options ?

### ISSUE THREAD

**Issue by juanmafont** [calamares/calamares#1439]
I am a user of Manjaro XFCE that uses Calamares. My notebook is "special", it's a chromebook with an intel video integrated card. So I need add always some parameters to (kernel) grub.

Will be possible add this option to Calamares (for advanced users, maybe a tab or button or other option to add default kernel parameters)

  Thanks a lot for your great work

**Comment by demmm** [comment]
>I need mount a chroot system to add kernel parameters

Both GRUB and systemd-boot will let you edit the kernel parameters in their menu, so not clear why you are chrooting.  Once you edited on first reboot, you can just adjust grub/systemd-boot entries.

Just an FYI, no opinion whether this editing should be possible within Calamares.

**Comment by adriaandegroot** [comment]
Calamares is a very linear installer: it goes step-by-step. I can vaguely imagine a "tweaks" viewstep, but it will look weird to a lot of users. If you can propose a spot in the UI for "edit kernel parameters" that makes sense outside of a tweak page, then this feature-request becomes a lot more feasible. Pop-up dialogs are ok, I suppose.

Related is "how to get these parameters to the right spot in Calamares"; if it's the bootloader module (it does some of the grub-wrangling) then it needs to be told where (in GlobalStorage, since you're moving information between modules) the special parameters live.

So I'm not closing this, but it's very much a "send PR" kind of issue to me (perhaps also for *calamares-extensions* repo, as a specialized module)

**Comment by juanmafont** [comment]
> > I need mount a chroot system to add kernel parameters
> 
> Both GRUB and systemd-boot will let you edit the kernel parameters in their menu, so not clear why you are chrooting. Once you edited on first reboot, you can just adjust grub/systemd-boot entries.
> 
> Just an FYI, no opinion whether this editing should be possible within Calamares.

Not if kernel parameters are needed to see you screen. I got blank screen when boot without this kernel boot parameters:

 i915.fastboot= 1 and i915.modeset=1

 When boot from live pendrive I pass this parameters at first step, then before reboot (after Calamares do his work) I need a chroot to run 'update-grub'.

**Comment by juanmafont** [comment]
> Calamares is a very linear installer: it goes step-by-step. I can vaguely imagine a "tweaks" viewstep, but it will look weird to a lot of users. If you can propose a spot in the UI for "edit kernel parameters" that makes sense outside of a tweak page, then this feature-request becomes a lot more feasible. Pop-up dialogs are ok, I suppose.
> 
> Related is "how to get these parameters to the right spot in Calamares"; if it's the bootloader module (it does some of the grub-wrangling) then it needs to be told where (in GlobalStorage, since you're moving information between modules) the special parameters live.
> 
> So I'm not closing this, but it's very much a "send PR" kind of issue to me (perhaps also for _calamares-extensions_ repo, as a specialized module)

Of course, you are right, the principle KISS is good, and Calamares and his linear step by step  pass is a good way to do his work. Maybe extension is the good solution.
By the way, I did not know about the existence of extensions for Calamares.

Thanks.

**Comment by abucodonosor** [comment]
@juanmafont 

what @demmm meant is you can hit 'e' on grub and add those, hit F10 to boot, and then adjust whatever you wish.

Also for such known problems, your distro can easily help you, by having a post-install script to add just that for you either to grub or ad a module option to i915.

**Comment by juanmafont** [comment]
> @juanmafont
> 
> what @demmm meant is you can hit 'e' on grub and add those, hit F10 to boot, and then adjust whatever you wish.
> 
> Also for such known problems, your distro can easily help you, by having a post-install script to add just that for you either to grub or as a module option to i915.

My chromebook has not functions keys (Must be emulated after boot). And boot into text mode give me corrupted screen text (not readable)

**Comment by demmm** [comment]
>My chromebook has not functions keys
`e` is not a function key, it is the regular character (short for `edit`)

**Comment by juanmafont** [comment]
> > My chromebook has not functions keys
> 
> `e` is not a function key, it is the regular character (short for `edit`)

Oooooh, I don't  knows about key 'e'.... Fantastic, this solve my problem and avoid to do chroot.   

Thank a lot ....... I'm happy.

**Comment by kkofler** [comment]
The problem is that edits you make in the GRUB menu are not persistent, i.e., they only apply to that one boot, so you have to redo them each time you boot the system. You have to edit `grub.cfg` and/or `/etc/default/grub` (`grub-mkconfig` rewrites `grub.cfg` based on the settings in `/etc/default/grub`) to set the arguments permanently.

**Comment by demmm** [comment]
Since I kept answering here with both GRUB & systemd-boot in mind, I will add to the above that you need to edit `/boot/loader/entries/xxx.conf`, options line to make it permanent for systemd-boot

**Comment by abucodonosor** [comment]
> The problem is that edits you make in the GRUB menu are not persistent, i.e., they only apply to that one boot, so you have to redo them each time you boot the system. You have to edit `grub.cfg` and/or `/etc/default/grub` (`grub-mkconfig` rewrites `grub.cfg` based on the settings in `/etc/default/grub`) to set the arguments permanently.

Sure, but @juanmafont used to chroot into the installed system from the live iso to change that. With an edit, he can boot normal and edit grub files etc.
